controllers/spacebindingrequest/spacebindingrequest_controller_test.go

```diff
@@ -28,25 +36,62 @@ func TestCreateSpaceBindingRequest(t *testing.T) {
-		ctrl := newReconciler(t, hostClient, member1)
+		t.Run("spaceBinding doesn't exists it should be created", func(t *testing.T) {
+			// given
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t, sbr, sbrNamespace), "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t, janeSpace, janeMur, base1nsTier)
+			ctrl := newReconciler(t, hostClient, member1)
+
+			// when
+			_, err = ctrl.Reconcile(context.TODO(), requestFor(sbr))
+
+			// then
+			require.NoError(t, err)
+			// spaceBindingRequest exists with config and finalizer
+			spacebindingrequesttest.AssertThatSpaceBindingRequest(t, sbr.GetNamespace(), sbr.GetName(), member1.Client).
+				HasSpecSpaceRole("admin").
+				HasSpecMasterUserRecord(janeMur.Name).
+				HasConditions(spacebindingrequesttestcommon.Provisioning()).
+				HasFinalizer()
+			// there should be 1 spacebinding that was created from the SpaceBindingRequest
+			spacebindingtest.AssertThatSpaceBinding(t, test.HostOperatorNs, janeMur.Name, janeSpace.Name, hostClient).
+				Exists().
+				HasLabelWithValue(toolchainv1alpha1.SpaceBindingRequestLabelKey, sbr.GetName()). // check expected labels are there
+				HasLabelWithValue(toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey, sbr.GetNamespace())
+		})
 
-		// when
-		_, err = ctrl.Reconcile(context.TODO(), requestFor(sbr))
+		t.Run("spaceBinding exists and is up-to-date", func(t *testing.T) {
+			// given
+			spaceBinding := spacebindingtest.NewSpaceBinding(janeMur.Name, janeSpace.Name, "admin", sbr.GetName(), spacebindingtest.WithSpaceBindingRequest(sbr))
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t, sbr, sbrNamespace), "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t, janeSpace, janeMur, base1nsTier, spaceBinding)
+			ctrl := newReconciler(t, hostClient, member1)
+
+			// when
+			_, err = ctrl.Reconcile(context.TODO(), requestFor(sbr))
 
-		// then
-		require.NoError(t, err)
-		// spacebindingrequest exists with expected cluster roles and finalizer
-		spacebindingrequesttest.AssertThatSpaceBindingRequest(t, "jane-tenant", sbr.GetName(), member1.Client).
-			HasSpecSpaceRole("admin").
-			HasSpecMasterUserRecord("jane")
+			// then
+			require.NoError(t, err)
+			// spaceBindingRequest exists with config and finalizer
+			spacebindingrequesttest.AssertThatSpaceBindingRequest(t, sbr.GetNamespace(), sbr.GetName(), member1.Client).
+				HasSpecSpaceRole("admin").
+				HasSpecMasterUserRecord(janeMur.Name).
+				HasConditions(spacebindingrequesttestcommon.Ready()).
+				HasFinalizer()
+			// there should be 1 spacebinding that was created from the SpaceBindingRequest
+			spacebindingtest.AssertThatSpaceBinding(t, test.HostOperatorNs, janeMur.Name, janeSpace.Name, hostClient).Exists().
+				HasLabelWithValue(toolchainv1alpha1.SpaceBindingRequestLabelKey, sbr.GetName()). // check expected labels are there
+				HasLabelWithValue(toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey, sbr.GetNamespace())
```

Comment:
we expect that the labels should match the values in spec - this could be part of the `AssertThatSpaceBinding` function.

Reply:
I'm not sure I'm understanding this . The `AssertThatSpaceBinding` function is being used by other tests like `usersignup`, `spacebindingcleanup`. If I move this check inside the function it will not work there, but maybe I'm not understanding this correctly.